client.go

```diff
@@ -1,37 +1,39 @@
 package main
 
 import (
+	"context"
 	"fmt"
-	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
 	"os"
 	"strings"
 
-	"github.com/Azure/azure-sdk-for-go/profiles/2018-03-01/resources/mgmt/locks"
-	"github.com/Azure/azure-sdk-for-go/profiles/2018-03-01/resources/mgmt/resources"
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/hashicorp/go-azure-helpers/authentication"
-	`github.com/hashicorp/go-azure-helpers/sender`
+	"github.com/hashicorp/go-azure-helpers/sender"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/managementgroups/2021-04-01/managementgroups"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2020-05-01/managementlocks"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2022-09-01/resourcegroups"
+	"github.com/manicminer/hamilton/environments"
 )
 
 type AzureClient struct {
 	applicationsClient      *graphrbac.ApplicationsClient
+	authClient              *authentication.Config
 	groupsClient            *graphrbac.GroupsClient
-	locksClient             *locks.ManagementLocksClient
-	resourcesClient         *resources.GroupsClient
+	locksClient             *managementlocks.ManagementLocksClient
+	managementClient        *managementgroups.ManagementGroupsClient
+	resourcesClient         *resourcegroups.ResourceGroupsClient
 	servicePrincipalsClient *graphrbac.ServicePrincipalsClient
 	usersClient             *graphrbac.UsersClient
 }
 
 func buildAzureClient() (*AzureClient, error) {
 	environmentName := os.Getenv("ARM_ENVIRONMENT")
-	if environmentName == "" {
-		environmentName = azure.PublicCloud.Name
-	}
```

Comment:
don't we need this? otherwise we require folks to explicitly specify this?

Reply:
I actually get an error if this is left in because of this function:
<img width="705" alt="image" src="https://user-images.githubusercontent.com/25562180/214066226-9adcd6e1-a3d5-4e8f-9fce-8c2233dae605.png">
Leaving the env var unset passes an empty string to that function which returns the desired default.